### DISCUSSION TITLE
Welcome to mitsubishi2MQTT Discussions!

### DISCUSSION THREAD

**Discussion by dzungpv** [dzungpv/mitsubishi2MQTT#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by avanh86** [comment]
Hi there im Alex and i am using a Mitsubishi HVAC 
Im willing to try out your config for it. And want to know if i can use an external temp sensor.
Also im struggeling with AsyncMqttClient to get installed in arduino.

**Comment by dzungpv** [comment]
> Hi there im Alex and i am using a Mitsubishi HVAC Im willing to try out your config for it. And want to know if i can use an external temp sensor. Also im struggeling with AsyncMqttClient to get installed in arduino.

Yes, you can, just push temperature data to this topic: **ha_remote_temp_set_topic = mqtt_topic + "/" + mqtt_fn + "/remote_temp/set"** .
About the libs, use my fork mention here: https://github.com/dzungpv/mitsubishi2MQTT/issues/5#issuecomment-1793469454

**Reply by avanh86** [reply]
I have tried to use the libs. But somehow im only getting error 13 codes. So for now im letting this go. I takes met too much time to get this working with arduino ide

**Reply by dzungpv** [reply]
> I have tried to use the libs. But somehow im only getting error 13 codes. So for now im letting this go. I takes met too much time to get this working with arduino ide

Just download the bin file build by github action, or using platformio to build, it is super easy and manage all the libs for you.

**Comment by BoulderM** [comment]
Great fork! Seems to work better than using ESPHome with SwiCago HeatPump library's.

Is it possible to add config options to add Wifi credentials and a static IP before uploading? I'm looking for a way to bulk upload to a couple of ESP devices.

**Reply by dzungpv** [reply]
It is not support static ip and wifi info like esphome, you can open the captivate portal and set it, save and unplug the power.

**Comment by Ta-Raffy** [comment]
Hey, nice work!
The most reliable Mitsubishi2MQTT I could find so far.

I just have a recurring problem with the last and current versions.

After restarting the WiFi network, only 4-5 of 8 ESP8266 connect to the MQTT again.
They connect to the WiFi, so I haven't been able to find any errors so far.

Is this a general Problem?
I'm now using a watchdog called /http://xxx.xxx.xxx.xx/?REBOOT=
with MQTT offline status, it would of course be better if a reconnect were triggered from the ESP

Otherwise really great! I would be happy to receive feedback.

Best regards
Raffy

**Reply by dzungpv** [reply]
You can wait 5 minutes and it will reconnect again, the logic add to prevent you must re flash device when enter wrong SSID/password. You can make it shorter by change the: https://github.com/dzungpv/mitsubishi2MQTT/blob/8ead953150d810e42719fbc0d3eb6e841284674a/main/config.h#L218C24-L218C46
If you reboot it or after power outage it will reconnect immediately. You can create new issue if it still occur after you power cycle the HP/ESP

**Reply by Ta-Raffy** [reply]
Hey Thanks for your reply , with the 2023.12.3 the issue is fixed 👍 
No Problems with connection 24/7 anymore.

Thank you 
Best Regards

Raffy

**Comment by PantheraIgnis** [comment]
This is the first way i try that actually works, thank you very much!
Is it just me or does it show some odd behavior when setting the mode, jumping to different values for the room temperature?

**Comment by p-ranit** [comment]
Hi! 
I am using Mitsubishi2MQTT firmware from gysmo38. Due to the argument/power/power on issue, I would like to test your fork. I tried to upload the bin file (Wemos version) from this repository to the "Firmware Upgrade" webpage of the device, but it reports an error during the upload (File upload buffer miscompare). The Wemos board is inside the HAVC unit and I would prefer not to disassemble it for testing. Is there any possibility to upload this firmware to the board without disassembling the HAVC? Thanks.

**Reply by dzungpv** [reply]
> Hi! I am using Mitsubishi2MQTT firmware from gysmo38. Due to the argument/power/power on issue, I would like to test your fork. I tried to upload the bin file (Wemos version) from this repository to the "Firmware Upgrade" webpage of the device, but it reports an error during the upload (File upload buffer miscompare). The Wemos board is inside the HAVC unit and I would prefer not to disassemble it for testing. Is there any possibility to upload this firmware to the board without disassembling the HAVC? Thanks.

What chip do you using, ESP8266 or ESP32?

**Reply by dzungpv** [reply]
> Hi! I am using Mitsubishi2MQTT firmware from gysmo38. Due to the argument/power/power on issue, I would like to test your fork. I tried to upload the bin file (Wemos version) from this repository to the "Firmware Upgrade" webpage of the device, but it reports an error during the upload (File upload buffer miscompare). The Wemos board is inside the HAVC unit and I would prefer not to disassemble it for testing. Is there any possibility to upload this firmware to the board without disassembling the HAVC? Thanks.

Remember to exact the zip file and upload the .bin file only. I try Webmos D1 minit build and it can be upload to the firmware page

**Reply by p-ranit** [reply]
> > Hi! I am using Mitsubishi2MQTT firmware from gysmo38. Due to the argument/power/power on issue, I would like to test your fork. I tried to upload the bin file (Wemos version) from this repository to the "Firmware Upgrade" webpage of the device, but it reports an error during the upload (File upload buffer miscompare). The Wemos board is inside the HAVC unit and I would prefer not to disassemble it for testing. Is there any possibility to upload this firmware to the board without disassembling the HAVC? Thanks.
> 
> What chip do you using, ESP8266 or ESP32?

Thanks for the quick reply. The chip is an ESP8266, as you can see from the attached image. 
![20240330](https://github.com/user-attachments/assets/c8457113-da2f-4b43-a04c-970e2a70f30f)

**Reply by p-ranit** [reply]
> > Hi! I am using Mitsubishi2MQTT firmware from gysmo38. Due to the argument/power/power on issue, I would like to test your fork. I tried to upload the bin file (Wemos version) from this repository to the "Firmware Upgrade" webpage of the device, but it reports an error during the upload (File upload buffer miscompare). The Wemos board is inside the HAVC unit and I would prefer not to disassemble it for testing. Is there any possibility to upload this firmware to the board without disassembling the HAVC? Thanks.
> 
> Remember to exact the zip file and upload the .bin file only. I try Webmos D1 minit build and it can be upload to the firmware page

I first tried the .bin file inside https://github.com/dzungpv/mitsubishi2MQTT/releases/download/2025.06.15/mitsubishi2MQTT_WEMOS_D1_Mini.bin.zip and now I also tried the .bin file inside https://github.com/dzungpv/mitsubishi2MQTT/releases/download/2025.06.15/mitsubishi2MQTT_ESP8266-ESP01.bin.zip, but in both cases the error is the same ("File upload buffer miscompare Upload error code 4").

I noticed that the .bin file in the gysmo38 version is smaller than yours. 
I am currently using the "Mitsubishi2MQTT 2023.8.0" version. 
There is also a newer version (2024.08). Maybe I should update to the latest version first and then try to migrate to your firmware. Have you tried updating from the gysmo38 firmware as well? What chip and gysmo38 firmware version were you using?
Thanks for the help

**Reply by dzungpv** [reply]
> > > Hi! I am using Mitsubishi2MQTT firmware from gysmo38. Due to the argument/power/power on issue, I would like to test your fork. I tried to upload the bin file (Wemos version) from this repository to the "Firmware Upgrade" webpage of the device, but it reports an error during the upload (File upload buffer miscompare). The Wemos board is inside the HAVC unit and I would prefer not to disassemble it for testing. Is there any possibility to upload this firmware to the board without disassembling the HAVC? Thanks.
> > 
> > 
> > Remember to exact the zip file and upload the .bin file only. I try Webmos D1 minit build and it can be upload to the firmware page
> 
> I first tried the .bin file inside https://github.com/dzungpv/mitsubishi2MQTT/releases/download/2025.06.15/mitsubishi2MQTT_WEMOS_D1_Mini.bin.zip and now I also tried the .bin file inside https://github.com/dzungpv/mitsubishi2MQTT/releases/download/2025.06.15/mitsubishi2MQTT_ESP8266-ESP01.bin.zip, but in both cases the error is the same ("File upload buffer miscompare Upload error code 4").
> 
> I noticed that the .bin file in the gysmo38 version is smaller than yours. I am currently using the "Mitsubishi2MQTT 2023.8.0" version. There is also a newer version (2024.08). Maybe I should update to the latest version first and then try to migrate to your firmware. Have you tried updating from the gysmo38 firmware as well? What chip and gysmo38 firmware version were you using? Thanks for the help

You can download file mitsubishi2MQTT_WEMOS_D1_Mini.bin in this link: https://github.com/dzungpv/mitsubishi2MQTT/actions/runs/15665446460 and it will work.

**Reply by p-ranit** [reply]
Thank you so much for the support. The problem was not your firmware version, but the previous one of gysmo38, the 2023.08.
I updated to its version 2024.08 and then to the version you linked me and now everything works correctly.
In the next few days I will test everything.

Sorry for the wasted time.
Kind regards

**Comment by ghqam** [comment]
Great job, congratulations!
I used an ESP8266-12F board, flashed with your latest release bin file, and everything works, but I'm facing a couple of minor issues:
- I can't enable the password requirement to control the device from the web page. I entered my password, but access remains free. I also tried using another browser and in private browsing.
- The other problem is that often, after sending a command (on, off, or whatever) to the air conditioner, the board seems to disconnect from the air conditioner, and it will reconnect after some seconds. In this case, when accessing the status page, the HVAC appears disconnected.

**Reply by dzungpv** [reply]
> Great job, congratulations! I used an ESP8266-12F board, flashed with your latest release bin file, and everything works, but I'm facing a couple of minor issues:
> 
>     * I can't enable the password requirement to control the device from the web page. I entered my password, but access remains free. I also tried using another browser and in private browsing.
> 
>     * The other problem is that often, after sending a command (on, off, or whatever) to the air conditioner, the board seems to disconnect from the air conditioner, and it will reconnect after some seconds. In this case, when accessing the status page, the HVAC appears disconnected.

I test it has no problem with login. Try to download latest file here and ota over web: https://github.com/dzungpv/mitsubishi2MQTT/actions/runs/16558770842 Using MQTT is more stable then web UI.

**Reply by ghqam** [reply]
Thanks for your answer,
I flashed the device OTA and I noticed a change in appearance, dark mode was activated.
But, unfortunately, the authentication request for HTTP access still doesn't work. And I've discovered that the connection loss that occurs after sending commands to the HVAC is caused by a reboot of the board. In fact, checking the information page, I see that the device rebooted when it wasn't responding.
I will try to use MQTT, but I don't think that this will fix the reboot problem.

**Reply by dzungpv** [reply]
> Thanks for your answer, I flashed the device OTA and I noticed a change in appearance, dark mode was activated. But, unfortunately, the authentication request for HTTP access still doesn't work. And I've discovered that the connection loss that occurs after sending commands to the HVAC is caused by a reboot of the board. In fact, checking the information page, I see that the device rebooted when it wasn't responding. I will try to use MQTT, but I don't think that this will fix the reboot problem.

Where do you buy the board? @ollo69  do you have this problem?

**Reply by ghqam** [reply]
This is my board https://www.az-delivery.de/en/products/d1-mini and I bought it from Amazon IT.
I have 2 more, in the next few days I will mount another one on another HVAC.
But the authentication problem is already visible now that I flashed it and it is not yet connected to the HVAC.
We'll see if this one too will suffer from the unexpected reboot.

**Reply by ollo69** [reply]
> But, unfortunately, the authentication request for HTTP access still doesn't work. 

Never used HTTP authentication in my config

>And I've discovered that the connection loss that occurs after sending commands to the HVAC is caused by a reboot of the board. In fact, checking the information page, I see that the device rebooted when it wasn't responding.
> I will try to use MQTT, but I don't think that this will fix the reboot problem.

Using HTTP interface to control HVAC device on WemosD1 cause reboot very often. This is due to very limited WemosD1 memory  and huge memory used by that page, that generate memory fragmentation and reboot after some refresh. In the end, I abandoned the WemosD1 and migrated to the ESP32, which now has decidedly similar prices but significantly superior performance. However, using the WemosD1 only via MQTT doesn't cause any problems.

**Reply by dzungpv** [reply]
> This is my board https://www.az-delivery.de/en/products/d1-mini and I bought it from Amazon IT. I have 2 more, in the next few days I will mount another one on another HVAC. But the authentication problem is already visible now that I flashed it and it is not yet connected to the HVAC. We'll see if this one too will suffer from the unexpected reboot.

You can download file here and update and test: https://github.com/dzungpv/mitsubishi2MQTT/actions/runs/16669177676 I have migrate to new ESP32Async/ESPAsyncWebServer Login should work now. The old version of the web server may cause the problem, but better using MQTT or you can try use ESP32/C3/S2/S3 with more processing power and ram.

**Reply by dzungpv** [reply]
> > But, unfortunately, the authentication request for HTTP access still doesn't work.
> 
> Never used HTTP authentication in my config
> 
> > And I've discovered that the connection loss that occurs after sending commands to the HVAC is caused by a reboot of the board. In fact, checking the information page, I see that the device rebooted when it wasn't responding.
> > I will try to use MQTT, but I don't think that this will fix the reboot problem.
> 
> Using HTTP interface to control HVAC device on WemosD1 cause reboot very often. This is due to very limited WemosD1 memory and huge memory used by that page, that generate memory fragmentation and reboot after some refresh. In the end, I abandoned the WemosD1 and migrated to the ESP32, which now has decidedly similar prices but significantly superior performance. However, using the WemosD1 only via MQTT doesn't cause any problems.

I have changed to ESP32 too, still think about archived ESP8266 code
